src/ucs/datastruct/bitmap.h

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_BITMAP_H_
+#define UCS_BITMAP_H_
+
+#include <ucs/sys/compiler_def.h>
+#include <stdint.h>
+
+BEGIN_C_DECLS
+
+
+typedef uint64_t ucs_bitmap_word_t;
+
+
+#define UCS_BITMAP_BITS_IN_WORD (sizeof(ucs_bitmap_word_t) * 8)
+
+
+#define UCS_BITMAP_WORD_INDEX(_bit_index) ((_bit_index) / UCS_BITMAP_BITS_IN_WORD)
+
+
+#define _UCS_BITMAP_BIT_IN_WORD_INDEX(_bit_index) ((_bit_index) % \
+                                                   UCS_BITMAP_BITS_IN_WORD)
+
+
+#define _UCS_BITMAP_BITS_TO_WORDS(_length) \
+    ((_length) / UCS_BITMAP_BITS_IN_WORD + \
+     ((_length) % UCS_BITMAP_BITS_IN_WORD ? 1 : 0))
+
+
+#define _UCS_BITMAP_WORD(_bitmap, _index) \
+    ((_bitmap)->bits[UCS_BITMAP_WORD_INDEX(_index)])
+
+
+#define _UCS_BITMAP_WORD_INDEX0(_bit_index) \
+    ((_bit_index) & ~(UCS_BITMAP_BITS_IN_WORD - 1))
+
+
+/*
+ * Represents an n-bit bitmap, by using an array
+ * of 64-bit unsigned long integers.
+ * 
+ * @param _length Number of bits in the bitmap
+ */
+#define _UCS_BITMAP_DECLARE_TYPE(_length) \
+    typedef struct { \
+        ucs_bitmap_word_t bits[_UCS_BITMAP_BITS_TO_WORDS(_length)]; \
+    } ucs_bitmap_t(_length);
+
+
+/**
+ * Expands to bitmap type definition
+ *
+ * @param _length Number of bits (as passed to UCS_BITMAP_DECLARE_TYPE)
+ *
+ * Example:
+ *
+ * @code{.c}
+ * ucs_bitmap_t(64) my_bitmap;
+ * @endcode
+ */
+#define ucs_bitmap_t(_length) ucs_bitmap_##_length##_suffix
+    
+
+_UCS_BITMAP_DECLARE_TYPE(64)
+_UCS_BITMAP_DECLARE_TYPE(128)
+_UCS_BITMAP_DECLARE_TYPE(256)
+
+
+/**
+ * Get the value of a bit in the bitmap
+ * 
+ * @param _bitmap Read value from this bitmap
+ * @param _index  Bit index to read
+ * 
+ * @return Bit value (0 or 1)
+*/
+#define UCS_BITMAP_GET(_bitmap, _index) \
+    (!!(_UCS_BITMAP_WORD(_bitmap, _index) & \
+    UCS_BIT(_UCS_BITMAP_BIT_IN_WORD_INDEX(_index))))
+
+
+/**
+ * Set the value of a bit in the bitmap
+ * 
+ * @param _bitmap Set value in this bitmap
+ * @param _index  Bit index to set
+ */
+ #define UCS_BITMAP_SET(_bitmap, _index) \
+    (_UCS_BITMAP_WORD(_bitmap, _index) |= \
+        (UCS_BIT(_UCS_BITMAP_BIT_IN_WORD_INDEX(_index))));
```

Comment:
indentation looks strange

Reply:
Indeting +4 after line break on `=` - I tried to be consistent with other code, for example:
https://github.com/openucx/ucx/blob/master/src/ucs/datastruct/array.h#L111